packages/apm/src/integrations/tracing.ts

```diff
@@ -778,7 +712,11 @@ function xhrCallback(handlerData: { [key: string]: any }): void {
   if (activity) {
     const span = activity.span;
     if (span && handlerData.xhr.setRequestHeader) {
-      handlerData.xhr.setRequestHeader('sentry-trace', span.toTraceparent());
+      try {
+        handlerData.xhr.setRequestHeader('sentry-trace', span.toTraceparent());
+      } catch (_) {
+        // Error: InvalidStateError: Failed to execute 'setRequestHeader' on 'XMLHttpRequest': The object's state must be OPENED.
```

Comment:
Is this intended or debugging?

Reply:
I encountered this while testing wasn't breaking anything but spamming the log.
This was I think when we cancel requests in the UI when navigating.

Reply:
Hmm, maybe the fix here is to only set the header before the request is made?

Reply:
Let's fix this in another PR

Reply:
Tracking in https://github.com/getsentry/sentry-javascript/issues/2505